### DISCUSSION TITLE
Api routes with cors

### DISCUSSION THREAD

**Discussion by clement-faure** [vercel/next.js#15395] [Help]
According to 

- the documentation : https://nextjs.org/docs/api-routes/api-middlewares
- this exemple repository :  https://github.com/vercel/next.js/tree/canary/examples/api-routes-cors

An error should be throw when the method is not allowed by cors (when using cors middleware on API routes).

I can not implement this feature on my side, i want to create a "POST" only route but i can still execute GET..

<img width="854" alt="Capture d’écran 2020-07-22 à 09 57 38" src="https://user-images.githubusercontent.com/23209488/88150528-ccba0c80-cc01-11ea-9bb8-24bf417f05ec.png">

Thank you in advance for your help !

**Comment by clement-faure** [comment]
I'll handle it using custom function for now, but yeah i don't understand what happening

![image](https://user-images.githubusercontent.com/23209488/88150881-3f2aec80-cc02-11ea-8c48-48b7896258fd.png)

**Comment by bakerue-owner** [comment]
Ive run into the exact same problem not sure how to fix it
